mmcv/runner/base_module.py

```diff
@@ -38,12 +54,33 @@ def is_init(self):
 
     def init_weights(self):
         """Initialize the weights."""
-        from ..cnn import initialize
 
+        # judge if it is Topmost module
+        top_most_module = len(self.params_init_info) == 0
+        if top_most_module:
+            for name, param in self.named_parameters():
+                self.params_init_info[param]['params_name'] = name
+                self.params_init_info[param][
+                    'init_info'] = 'The value is the same as ' \
+                                   'the default initialization of PyTorch'
+                self.params_init_info[param]['tmp_sum_value'] = param.data.sum(
+                )
+            # pass `params_init_info` to all submodules
```

Comment:
May add more comments here, indicating that: all submodules will modify the same params_init_info during initialization thus params_init_info will keep the final initialization information of each module.